lib/core/src/sdk.rs

```diff
@@ -701,7 +701,7 @@ impl LiquidSdk {
             InputType::LiquidAddress {
                 address: mut liquid_address_data,
             } => {
-                let Some(amount_sat) = liquid_address_data.amount_sat else {
+                let Some(amount_sat) = liquid_address_data.amount_sat.or(req.amount_sat) else {
```

Comment:
Shouldn't we give more priority to the amount the user put in the request rather than in the bip21 input?

Reply:
I think the same as a user can pay different amount in lightning for an bolt11 invoice that has a specific amount, we should allow here the user to override the bip21 amount (as nothing can enforce that amount really) and pay different amount. 
So IMO the amount in the request (if exists) takes priority over the amount in the bip21 as it was explicitly set by the user.